pkg/codec/config.go

```diff
@@ -117,30 +125,51 @@ func (d *DropModifierConfig) ToModifier(onChainHooks ...mapstructure.DecodeHookF
 	return NewRenamer(fields), nil
 }
 
-// ElementExtractorConfig is used to extract an element from a slice or array
-type ElementExtractorConfig struct {
+func (d *DropModifierConfig) MarshalJSON() ([]byte, error) {
+	return json.Marshal(&modifierMarshaller[DropModifierConfig]{
+		Type: ModifierDrop,
+		T:    d,
+	})
+}
+
+// ElementExtractorModifierConfig is used to extract an element from a slice or array
+type ElementExtractorModifierConfig struct {
 	// Key is the name of the field to extract from and the value is which element to extract.
 	Extractions map[string]*ElementExtractorLocation
 }
 
-func (e *ElementExtractorConfig) ToModifier(onChainHooks ...mapstructure.DecodeHookFunc) (Modifier, error) {
+func (e *ElementExtractorModifierConfig) ToModifier(_ ...mapstructure.DecodeHookFunc) (Modifier, error) {
```

Comment:
I'm curious, why are we ignoring the hook functions for drop & extract, but not for hard code?

Reply:
dropping/rename change the name in an underlying struct.  They don't actually do anything with maps. 

extracting one takes an element from the value.  It doesn't need to know anything about how your types are made.  Your type is already there, or will later be converted at that point.

Hard-code will need to know how to sometimes convert types.  For example, an address is a string in config, but it doesn't if it's a hex encoding, base64 encoding, or something else.  Big int can overflow integers in json, so you may need to quote the value, it needs to have the big.Int hook, etc.